common/page.go

```diff
@@ -153,6 +171,48 @@ func NewPage(
 	return &p, nil
 }
 
+func (p *Page) initEvents() {
+	p.logger.Debugf("NewPage:initEvents",
+		"sid:%v tid:%v", p.session.ID(), p.targetID)
+
+	events := []string{
+		cdproto.EventRuntimeConsoleAPICalled,
+	}
+	p.session.on(p.ctx, events, p.eventCh)
+
+	go func() {
+		p.logger.Debugf("NewPage:initEvents:go",
+			"sid:%v tid:%v", p.session.ID(), p.targetID)
+		defer func() {
+			p.logger.Debugf("NewPage:initEvents:go:return",
+				"sid:%v tid:%v", p.session.ID(), p.targetID)
+			// TaskQueue is only initialized when calling page.on() method
+			// so users are not always required to close the page in order
+			// to let the iteration finish.
+			if p.tq != nil {
+				p.tq.Close()
```

Comment:
In regards of requiring to close the page so the `TaskQueue` is closed and therefore the iteration can end. I've changed this in [60a94f0](https://github.com/grafana/xk6-browser/pull/1006/commits/60a94f0b3a4c9f0be2826bb62eef47a583ffa422) so the `TaskQueue` is only initialized when calling `page.on()` method, and therefore only has to be closed when using this method. I think this is a better way, to not introduce more breaking changes by now, and this particular case can be documented in the `page.on` method description.
Also I think requiring the user to close the page only when using this method makes more sense, as `page.on` defines a function that will be executed for every event (in this case only `console` events) throughout the page's life, so it's easier to understand that the page has to be closed so the handler execution stops and the iteration can finish.

WDYT @inancgumus @ankur22 ?

Reply:
I think this is an excellent idea from the usability perspective 👍 As you suggested, we can mention this in a notice box in the documentation 👍